exporters/prometheus/src/prometheus_exporter_utils.cc

```diff
@@ -0,0 +1,467 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <sstream>
+#include <utility>
+#include <vector>
+
+#include "opentelemetry/exporters/prometheus/prometheus_exporter_utils.h"
+#include "opentelemetry/sdk/metrics/aggregator/aggregator.h"
+#include "prometheus/metric_type.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace prometheus
+{
+/**
+ * Helper function to convert OpenTelemetry metrics data collection
+ * to Prometheus metrics data collection
+ *
+ * @param records a collection of metrics in OpenTelemetry
+ * @return a collection of translated metrics that is acceptable by Prometheus
+ */
+std::vector<prometheus_client::MetricFamily> PrometheusExporterUtils::TranslateToPrometheus(
+    std::vector<metric_sdk::Record> &records)
+{
+  if (records.empty())
+  {
+    return {};
+  }
+
+  // initialize output vector
+  std::vector<prometheus_client::MetricFamily> output(records.size());
+
+  // iterate through the vector and set result data into it
+  int i = 0;
+  for (auto r : records)
+  {
+    SetMetricFamily(r, &output[i]);
+    i++;
+  }
+
+  return output;
+}
+
+// ======================= private helper functions =========================
+/**
+ * Set value to metric family according to record
+ */
+void PrometheusExporterUtils::SetMetricFamily(metric_sdk::Record &record,
```

Comment:
Can record be a constref?

Reply:
If I make record a constref here, I need to change all getter functions in Record class const functions. Record class was implemented by previous interns, and also used in other classes.